src/modules/Signup.tsx

```diff
@@ -1,60 +1,77 @@
 import { useState } from 'react';
 import { Link, useNavigate } from "react-router-dom";
 import { Formik, Form, Field, ErrorMessage } from "formik";
-import { signupValidationSchema } from "../validations/validation";
+import * as yup from 'yup';
 import axios from 'axios';
 import { User } from '../types/type';
+import { toast } from "react-toastify";
 
 const Signup = () => {
   const [formData, setFormData] = useState({
-    ID:0,
+    ID: 0,
     name: "",
     email: "",
     password: "",
     phone_number: "",
     user_type: "",
-    
   });
 
   const navigate = useNavigate();
 
-
-  const handleSubmit = async (values:User) => {
+  const signupValidationSchema = yup.object({
+    name: yup.string()
+      .required("Name is required")
+      .matches(/^[^\s]+$/, "Name must not contain spaces"),
+    email: yup.string().email("Invalid email format").required("Email is required"),
+    phone_number: yup.string()
+      .required("Phone number is required")
+      .matches(/^\d+$/, "Phone number must contain only digits"),
+    user_type: yup.string().required("User type is required"),
+    password: yup.string()
+      .required("Password is required")
+      .min(8, "Password must be at least 8 characters long")
+      .matches(
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
+        "Password must contain at least one uppercase letter, one lowercase letter, one number, and one special character"
+      ),
+  });
+  
+  const handleSubmit = async (values: User) => {
     try {
-      const response = await axios.post('http://localhost:8080/users',{ ...values,phone_number: parseInt(values.phone_number)});
+      const response = await axios.post('http://localhost:8080/users', { ...values, phone_number: parseInt(values.phone_number) });
```

Comment:
Again we never do get/post/put actions directly in react code. You should always follow the RTK query or Redux sagas etc asynchronous flow